### PR TITLE
For discussion: Fix sentinels for live playback where time goes "backwards"

### PR DESCRIPTION
**Pull request by dhurrell**
Our original spike to avoid the buffering sentinel being activated when time goes "backwards" wasn't quite right. The buffering sentinel was no longer being activated, but the seek sentinel was being activated instead.

This pull request attempts to fix both problems, but requires a little bit of discussion because I know we were treating the _sentinelSeekTime member variable in the original way for a reason.

### CODE REVIEW

File: static/script/devices/mediaplayer/html5.js [hunk 1]
```diff
@@ -587,7 +587,7 @@ require.def(
                         self._mediaElement.currentTime = self._sentinelSeekTime;
                     });
                 } else {
-                    this._sentinelSeekTime = currentTime;
+                    this._sentinelSeekTime = undefined;
```
Comment (sdclibbery) [inline static/script/devices/mediaplayer/html5.js]: As you say, David, I'm sure there was a reason for this. Why is this change needed for this pull request? :-)

Comment (dhurrell) [inline static/script/devices/mediaplayer/html5.js]: If we don't reset this member variable, then when time goes "backwards" by more than 15 seconds (the tolerance used by the seek sentinel), the seek sentinel activates. Although this isn't the bug we originally set out to fix, it seems just as much of a problem as the original issue, which was the buffering sentinel kicking in where time goes "backwards" by <15s.

An alternative solution could be to say that any device that makes time go "backwards" while playing live media cannot use the "seekable" live modifier, and we could have the other modifiers disable the seek sentinel.

Comment (dhurrell) [inline static/script/devices/mediaplayer/html5.js]: Remembered the reason, I think: it was because we'd observed a device where the time would temporarily be reported as the target time for the seek, then skip back to its original position. There can't be a unit test for such a scenario, otherwise this pull request would be breaking the build.

Does that sound right?

Comment (sdclibbery) [inline static/script/devices/mediaplayer/html5.js]: Hm, the previous behaviour here of setting _sentinelSeekTime to the current time keeps the sentinel active, but also keeps the target time 'rolling forward' as playback continues.

Setting _sentinelSeekTime to undefined will effectively disable the seek sentinel forever (well, at least until the next playFrom); see line 577. So, doesn't this change mean that the seek sentinel will never be able to work on any device in any circumstances? :-)

Comment (dhurrell) [inline static/script/devices/mediaplayer/html5.js]: Well, my thinking was that the seek sentinel had done its job at this point, so it could safely be disabled until the next playFrom(). But we know that some devices report the target seek position as the current time, then hop back to the original time, so this won't actually work.

Since this pull request hasn't broken any unit tests, we obviously have a gap in our testing. I'm going to add a unit test for the seek-then-hop-back scenario before going any further.

File: static/script/devices/mediaplayer/html5.js [hunk 2]
```diff
@@ -642,12 +644,12 @@ require.def(
                 var self = this;
                 this._clearSentinels();
                 this._sentinelIntervalNumber = 0;
-                this._sentinelSetTime = this.getCurrentTime();
```
Comment (dhurrell) [inline static/script/devices/mediaplayer/html5.js]: This variable is unused.

**Comment by sdclibbery** [comment]
So, testing indicated that the device was able to support live restart, and therefore we probably want to keep this entire change.
